app-modules/ai/src/Services/Contracts/AiService.php

```diff
@@ -101,5 +117,19 @@ public function retryMessage(AiMessage $message, Closure $saveResponse): Closure
 
     public function getMaxAssistantInstructionsLength(): int;
 
+    public function retrieveFile(AiMessageFile $file): FilesDataTransferObject;
+
+    public function createVectorStore(array $parameters): VectorStoresDataTransferObject;
+
+    public function retrieveVectorStore(string $vectorStoreId): VectorStoresDataTransferObject;
+
+    public function modifyVectorStore(string $vectorStoreId, array $parameters): void;
+
+    public function retrieveVectorStoreFiles(AiThread $thread, string $vectorStoreId, array $params): VectorStoreFilesDataTransferObject;
```

Comment:
We may want to consider an additional interface called something like AiServiceWithVectorStores or something like that.

Not all AI services may have vector stores available. But we could also address this later when we encounter one that does not.

Thoughts @danharrin and @dgoetzit?

Reply:
These probably could just go into the `SupportsFileUploads` contract since the only reason we'd really want to interact with the stores is for the purpose of uploading/interacting with files.